x-pack/plugins/infra/public/pages/logs/log_entry_rate/sections/anomalies/log_entry_example.tsx

```diff
@@ -0,0 +1,279 @@
+import {
+  LogColumnConfiguration,
+  isTimestampLogColumnConfiguration,
+  isFieldLogColumnConfiguration,
+  isMessageLogColumnConfiguration,
+} from '../../../../../utils/source_configuration';
+import { localizedDate } from '../../../../../../common/formatters/datetime';
+
+export const exampleMessageScale = 'medium' as const;
+export const exampleTimestampFormat = 'time' as const;
+
+const MENU_LABEL = i18n.translate('xpack.infra.logAnomalies.logEntryExamplesMenuLabel', {
+  defaultMessage: 'View actions for log entry',
+});
+
+const VIEW_IN_STREAM_LABEL = i18n.translate(
+  'xpack.infra.logAnomalies.logEntryExamplesViewInStreamLabel',
+  {
+    defaultMessage: 'View in stream',
+  }
+);
+
+const VIEW_ANOMALY_IN_ML_LABEL = i18n.translate(
+  'xpack.infra.logAnomalies.logEntryExamplesViewAnomalyInMlLabel',
+  {
+    defaultMessage: 'View anomaly in machine learning',
+  }
+);
+
+type Props = LogEntryRateExample & {
+  timeRange: TimeRange;
+  jobId: string;
+};
+
+export const LogEntryRateExampleMessage: React.FunctionComponent<Props> = ({
+  id,
+  dataset,
+  message,
+  timestamp,
+  tiebreaker,
+  timeRange,
+  jobId,
+}) => {
+  const [isHovered, setIsHovered] = useState(false);
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const openMenu = useCallback(() => setIsMenuOpen(true), []);
+  const closeMenu = useCallback(() => setIsMenuOpen(false), []);
+  const setItemIsHovered = useCallback(() => setIsHovered(true), []);
+  const setItemIsNotHovered = useCallback(() => setIsHovered(false), []);
+
+  // the dataset must be encoded for the field column and the empty value must
+  // be turned into a user-friendly value
+  const encodedDatasetFieldValue = useMemo(
+    () => JSON.stringify(getFriendlyNameForPartitionId(dataset)),
+    [dataset]
+  );
+
+  const viewInStreamLinkProps = useLinkProps({
+    app: 'logs',
+    pathname: 'stream',
+    search: {
+      logPosition: encode({
+        end: moment(timeRange.endTime).format('YYYY-MM-DDTHH:mm:ss.SSSZ'),
+        position: { tiebreaker, time: timestamp },
+        start: moment(timeRange.startTime).format('YYYY-MM-DDTHH:mm:ss.SSSZ'),
+        streamLive: false,
+      }),
+      flyoutOptions: encode({
+        surroundingLogsId: id,
+      }),
+      logFilter: encode({
+        expression: `${partitionField}: ${dataset}`,
+        kind: 'kuery',
+      }),
+    },
+  });
+
+  const viewAnomalyInMachineLearningLinkProps = useLinkProps(
+    getEntitySpecificSingleMetricViewerLink(jobId, timeRange, {
+      [partitionField]: dataset,
+    })
+  );
+
+  const menuItems = useMemo(() => {
+    if (!viewInStreamLinkProps.onClick || !viewAnomalyInMachineLearningLinkProps.onClick) {
+      return undefined;
+    }
+
+    return [
+      {
+        label: VIEW_IN_STREAM_LABEL,
+        onClick: viewInStreamLinkProps.onClick,
+      },
+      {
+        label: VIEW_ANOMALY_IN_ML_LABEL,
+        onClick: viewAnomalyInMachineLearningLinkProps.onClick,
+      },
```

Comment:
1) 👍
2) Probably not, no. So, we can either move this to the row level (as you said). Or I guess the time range for the single metrics explorer link could be narrower so it sits more tightly around the anomaly time, rather than the page time range. Page time range works nicely for view in stream as we can still sit the exact line centrally, but I agree it doesn't work well for ML. Then again, I don't know if it's useful to have it as a starting point to jump in to for investigation?

Reply:
I'm unsure about the single metrics explorer link - let's discuss this as a follow-up (with @katrin-freihofner possibly)?

Reply:
👍